### ISSUE TITLE
When debugging tests (incl -p option): browser's developer tools closes constantly

### ISSUE THREAD

**Issue by czellweg** [Polymer/web-component-tester#242]
When I debug tests using (for example) Chrome's developer tools, it seems that some sort of refresh happens which constantly closes the developer tools after an indeterminate amount of seconds (sometimes less than 1s, sometimes up to 20s). I have to be quick enough to find out what I need in this time window which makes it a bit stressful. ;)

I haven't yet found out what causes this behaviour so maybe someone here has an idea?

I start `wct` with the following config:
```javascript
  module.exports = {
  verbose: false,
  persistent: true,
  plugins: {
    local: {
      browsers: ['chrome']
    }
  },
};
```

**Comment by czellweg** [comment]
I think I have just found the solution thanks to this blog post: http://japhr.blogspot.ch/2015/04/the-debugger-and-breakpoints-with-web.html

What worked for me was:
* start wct and copy the URL *minus* the `?cli_browser_id=0` argument
* open a new tab in the browser window that `wct` created and past that URL

Then, debugging should work as expected and the dev tools will remain open.

**Comment by bushuyev** [comment]
Could you please consider reopening this? Existing workaround doesn't mean it shouldn't be fixed, or, at least documented. It took me half of the day today to find this issue to solve it. Besides, it seems ot be related to 
https://sites.google.com/a/chromium.org/chromedriver/help/devtools-window-keeps-closing

**Comment by bushuyev** [comment]
and it is still closing.

**Comment by bushuyev** [comment]
Thank you!

**Comment by azakus** [comment]
Yeah, this is a complication from wct-sauce plugin, which sends a selenium command every 45 seconds to keep the sauce connection alive.

I need to refactor how the plugins load to fix this.

**Comment by bushuyev** [comment]
Sauce? I thought I've turned it off:


var wct = require('web-component-tester').test;
...
gulp.task('tests-persistent', ['systemjs'], function(){

  wct({persistent: true, plugins: {local: {}, sauce: false}, testTimeout: 99999999997}, function(error){
    if (error) {
      notifier.notify({
        'title': 'Build failed',
        'message': '' + error
      });
    }
    else {
      notifier.notify({'title': 'Build passed', 'message': 'Yay!'});
    }
  })
});

Have I?

**Comment by justinfagnani** [comment]
@azakus can you describe the refactoring you had in mind? Agree this is super annoying.

**Comment by azakus** [comment]
My thought was to not load the plugins unless they are specified on the command line or in the config file. Right now, wct will load both the local and sauce plugins even if the other is not used.

**Comment by justinfagnani** [comment]
In addition, is there a way to send the keep-alive command only to the sauce launched browsers?

**Comment by pdesjardins90** [comment]
+1 this is insane

**Comment by ankon** [comment]
FWIW, this still seems to happen with a disabled sauce plugin, and in debug mode I can see this output repeating - devtools close always at that time.
~~~~
chrome 58                CALL title()
chrome 58                GET /session/:sessionID/title
14:58:49.923 INFO - Executing: [get title])
14:58:49.927 INFO - Done: [get title]
chrome 58                RESPONSE title()
~~~~

**Comment by avizcaino** [comment]
Hi! Any news on this? I'm trying to debug some tests and tried everything here but devtools keeps closing... :(

**Comment by pdesjardins90** [comment]
It's workaround time!
- polymer test -p
- copy URL
- close chrome
- open a new chrome manually
- paste url + enter
- Voilà, devtools now stays open, but becomes more slow over time (seems like a memory leak), you'll have to reboot it after some time
